workshops/Image Sharing Site/README.md

```diff
@@ -0,0 +1,459 @@
+
+<img alt="Firestore is set" src="https://cloud-ilf6f1rk5.vercel.app/016.png">
+
+Now your firebase is set. Let's jump into coding.
+
+```html
+<input onchange="fileinput()" type="file" id="photo" />
+```
+
+Here, in the input tag, the onchange function is calling a function 'fileinput'. Let's make it.
+
+```js
+function fileinput(e) {
+  document.getElementById("status").innerHTML = "Setting up...";
+  const ref = firebase.storage().ref();
+  const file = document.querySelector("#photo").files[0];
+  const task = ref.child(file.name).put(file);
+  task
+    .then((snapshot) => snapshot.ref.getDownloadURL())
+    .then((url) => {
+      link = url;
+      document.getElementById("status").innerHTML = "Now you can upload it.";
+    });
+}
+```
+
+In this function, we are returning a value(here the upload status) to the h1 tag(#status) to display that the image is setting up. Then we are declaring a variable ref as firebase storage. Next, we declare another variable file as the input field file fetch(i.e, the image that we select to upload) The variable task is to put the file into the firebase storage.
+Then we are downloading the URL of the image that we just uploaded to our firebase storage.
+After that, we display that the image is uploaded so you can post it. This adds the URL we just downloaded to the firebase firestore.
+Let's push it to the firestore. This happens when you click the upload image button.
+
+```js
+function upload() {
+  const name = document.getElementById("name").value;
+  if (link === null) {
+    document.getElementById("status").innerHTML = "Select an image.";
+  } else if (name.length <= 2) {
+    document.getElementById("status").innerHTML =
+      "Name should be atleast 3 charector long.";
+  } else {
+    document.getElementById("status").innerHTML = "Uploading...";
+    const url = {
+      url: link,
+      name: name,
+    };
+    db.collection("pictures")
+      .doc()
+      .set(url)
+      .then(() => {
+        location.reload();
+      });
+  }
+}
+```
+
+We get the name the user gives in the input tag earlier and check if the URL of the image is null(meaning image not selected). Then we check whether the name is 3 characters long. After all this, we will display the message 'uploading...' and send our URL + name to our firestore. Now we need to display it right? Let's write the code for that.
+
+```js
+db.collection("pictures")
+  .get()
+  .then((querySnapshot) => {
+    const data = querySnapshot.docs.map((doc) => doc.data());
+    var a = 0;
+    for (var i in data) {
+      var img = document.createElement("img");
+      img.src = data[a].url;
+      img.setAttribute("width", "600");
+      var h = document.createElement("H1");
+      var t = document.createTextNode(data[a].name);
+      h.appendChild(t);
+      h.style.color = "deeppink";
+      document.getElementById("container").appendChild(h);
+      document.getElementById("container").appendChild(img);
+      a = a + 1;
```

Comment:
Can be simplified with `a++`

Reply:
(Yes Mathtew I know you told me to stay away hehe)

Or you could technically do ++a as well, my dad says it's more efficient.

Reply:
ok